### PR TITLE
[language-platform] move dependecies background jobs to its own layer

### PR DESCRIPTION
**Pull request by Numbers88s**
Move background jobs to its own layer within the service

Added dashboards for it.
CleanShot 2022-10-13 at 12 19 52@2x

Sidenote:
I want to move this to its own dashboard like the other services. Thoughts?

## Test plan
Update test and run them successfully

### CODE REVIEW

**Comment by efritz** [comment]
@Numbers88s This is just deps right? (Title + PR description don't make that obvious, was expecting a longer read 😅).

**Comment by Numbers88s** [comment]
> @Numbers88s This is just deps right? (Title + PR description don't make that obvious, was expecting a longer read 😅).

Yeah, 😅

**Comment by efritz** [comment]
> I want to move this to its own dashboard like the other services. Thoughts?

We should end up migrating everything out of frontend, worker, pci-worker, executor dashboards (only leaving infra-level monitoring) and have each of our dashboards defined by our domain (upload, codenav, autoindexing, etc). That way we can determine what's relevant to monitor once we define a handful of SLIs for each domain.

File: internal/codeintel/dependencies/internal/background/observability.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+type operations struct {
+	handleCrateSyncer *observation.Operation
+}
+
+func newOperations(observationContext *observation.Context) *operations {
+	m := metrics.NewREDMetrics(
+		observationContext.Registerer,
+		"codeintel_dependencies_background", // TODO: make sure to add.
```
Comment (efritz) [inline internal/codeintel/dependencies/internal/background/observability.go:17]: 👀

**Review by efritz — APPROVED** [review]

File: internal/codeintel/dependencies/internal/background/observability.go (hunk 1 above)
Comment (Numbers88s) [reply internal/codeintel/dependencies/internal/background/observability.go:17]: I did add it. Forgot to delete comment. :Facepalm:

**Pull request merged**